### PR TITLE
Added more REST examples to the documentation

### PR DESCRIPTION
**Pull request by jgweir**

### CODE REVIEW

File: end-user/en/source/pages/apis/using-apis.rst [hunk 1]
```diff
@@ -143,8 +160,462 @@ The query string is composed of one or more field-value pairs, each separated by
...
+	* API resource `osPkg_getAll <apis:osPkg_getAll>`_
+	* `Package object <apis:package-object>`_
+
+
+.. _apis-rest-add-os-pkg:
+
+Adding a Package to an Appliance's OS Profile (Linux Only)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once an Appliance Template has an OS profile, you can add or remove operating system packages to it.  To add one or more packages, you will need the URI of the packages(s) you wish to add.  You can :ref:`search <apis-rest-add-os-pkg-search>` for packages to retrieve this information.
```
Comment (emuus) [inline end-user/en/source/pages/apis/using-apis.rst]: I would make the link text "search for packages" (not just search)

File: end-user/en/source/pages/apis/using-apis.rst [hunk 2]
```diff
@@ -143,8 +160,462 @@ The query string is composed of one or more field-value pairs, each separated by
...
+
+Use the following API resource to add or remove these packages to the OS profile of an Appliance Template:
+
+.. function:: PUT /users/{uid}/appliances/{aid}/osprofile/{osid}/pkgs
+
+	* ``uid``: is the login of the user
+	* ``aid``: the id of the appliance template
+	* ``osid``: the id of the os profile where to add the packages
```
Comment (emuus) [inline end-user/en/source/pages/apis/using-apis.rst]: I would prefer : the id of the os profile to which the packages should be added

File: end-user/en/source/pages/apis/using-apis.rst [hunk 3]
```diff
@@ -143,8 +160,462 @@ The query string is composed of one or more field-value pairs, each separated by
...
+
+	1. A software component container is created.  This is registered into your software library.
+
+	2. Create a software artifact container.  This is the meta-date container for an uploaded file
+
+	3. Upload the software files into this software artifact container.
+
+	4. Add the software component to an appliance template.  Note, that this software component can be added to multiple appliance templates.
+
+To create the software component container, we use the following API resource:
```
Comment (emuus) [inline end-user/en/source/pages/apis/using-apis.rst]: remove "we"

File: end-user/en/source/pages/apis/using-apis.rst [hunk 4]
```diff
@@ -143,8 +160,462 @@ The query string is composed of one or more field-value pairs, each separated by
...
+The ``software.xml`` content (the request body):
+
+.. code-block:: xml
+
+	<ns0:mySoftware xmlns:ns0="http://www.usharesoft.com/uforge">
+	    <name>Zabbix</name>
+	    <version>3.0.1</version>
+	</ns0:mySoftware>
+
+Once created, note down the ``artifactsUri`` of this software component.  This is the URI we need to use to register one or more artifact objects.
```
Comment (emuus) [inline end-user/en/source/pages/apis/using-apis.rst]: replace "we" by "you"

File: end-user/en/source/pages/apis/using-apis.rst [hunk 5]
```diff
@@ -143,8 +160,462 @@ The query string is composed of one or more field-value pairs, each separated by
...
+
+Now we can create an artifact container.  To do this we use the ``uri`` of the software component.  The resource API is:
```
Comment (emuus) [inline end-user/en/source/pages/apis/using-apis.rst]: delete "we"

File: end-user/en/source/pages/apis/using-apis.rst [hunk 6]
```diff
@@ -143,8 +160,462 @@ The query string is composed of one or more field-value pairs, each separated by
...
+	    <fullName>
+	        zabbix-release-3.0-1.el6.noarch.rpm
+	        </fullName>
+	    <origName>
+	        zabbix-release-3.0-1.el6.noarch.rpm
+	        </origName>
+	    <subSoftwareArtifacts/>
+	</ns0:softwareFile>
+
+Now can actually upload the binary.  Note down the ``binaryUri`` of the newly created artifact object.  This is the resource uri we use to upload the file:
```
Comment (emuus) [inline end-user/en/source/pages/apis/using-apis.rst]: Now you can upload the binary (remove "actually" add "you"). Replace "we" by "you will" in last sentence

File: end-user/en/source/pages/apis/using-apis.rst [hunk 7]
```diff
@@ -143,8 +160,462 @@ The query string is composed of one or more field-value pairs, each separated by
...
+	* ``uid``: is the login of the user
+	* ``msid``: the id of the software compnent created
+	* ``said``: the id of the software artifact
+	* ``fileName`` (optional): The filename to upload
+
+Example (uploading a RPM, but this can be any file type)::
+
+	$ curl "http://10.1.2.206/api/users/guest/mysoftware/918/artifacts/1078/bin/" -X POST -H "Authorization: Basic guest:guest_password" -H "Accept: application/xml" --data-binary "/path/to/file/zabbix-release-3.0-1.el6.noarch.rpm" | tidy -xml -indent -quiet
+
+Finally we can now add this software component to an appliance template.  The following resource API is used:
```
Comment (emuus) [inline end-user/en/source/pages/apis/using-apis.rst]: replace "we" by "you"

File: end-user/en/source/pages/apis/using-apis.rst [hunk 8]
```diff
@@ -143,8 +160,462 @@ The query string is composed of one or more field-value pairs, each separated by
...
+
+. function:: PUT /users/{uid}/appliances/{aid}
```
Comment (emuus) [inline end-user/en/source/pages/apis/using-apis.rst]: need two periods?

File: end-user/en/source/pages/apis/using-apis.rst [hunk 9]
```diff
@@ -85,34 +85,51 @@ Request Example
...
+* For JSON, use: ``-H "Accept: application/json"``
+* For XML, use: ``-H "Accept: application/xml"``
+
+If no accept header is used, then ``XML`` is returned by default.
+
+For example, to retrieve the user information in JSON, the following request is used::
+
+	$ curl "http://10.1.2.206/api/users/guest" -X GET -u "guest:guest_password" -H "Accept: application/json"
+
+	.. note:: This example has used basic authentication (which is not advised).  Furthermore, the response body i.e. the user information has been omitted here for clarity.
```
Comment (jgweir) [inline end-user/en/source/pages/apis/using-apis.rst]: its fine :)

File: end-user/en/source/pages/apis/using-apis.rst [hunk 10]
```diff
@@ -143,8 +160,462 @@ The query string is composed of one or more field-value pairs, each separated by
 	http://server/uripath?field1=value1&field2=value2&field3=value3
 
 
+REST API Examples
+-----------------
+
+The following examples for clarity use basic authentication to communicate with UForge.  As  this is an insecure request, this is not recommended for production use.  All response information is also ommitted.
```
Comment (jgweir) [inline end-user/en/source/pages/apis/using-apis.rst]: ok

File: end-user/en/source/pages/apis/using-apis.rst [hunk 11]
```diff
@@ -143,8 +160,462 @@ The query string is composed of one or more field-value pairs, each separated by
...
+
+	* API resource `appliance_create <apis:appliance-create>`_
+	* `Appliance object <apis:appliance-object>`_
+
+.. _apis-rest-get-os-profiles:
+
+Retrieve Available OS Profiles for an Operating System
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Each appliance template has a notion of an OS profile.  This lists the operating system packages for the appliance template.  To help users to create OS profiles for an appliance template, each Operating System registered in the UForge platform has one one or more base OS profiles (also known as OS templates) the user can choose from to get started.  To retrieve the list of available OS profiles (or templates) for a particular operating system, use the following API resource:
```
Comment (jgweir) [inline end-user/en/source/pages/apis/using-apis.rst]: ok

File: end-user/en/source/pages/apis/using-apis.rst (hunk 1 above)
Comment (jgweir) [inline end-user/en/source/pages/apis/using-apis.rst]: ok

File: end-user/en/source/pages/apis/using-apis.rst (hunk 2 above)
Comment (jgweir) [inline end-user/en/source/pages/apis/using-apis.rst]: ok

File: end-user/en/source/pages/apis/using-apis.rst (hunk 3 above)
Comment (jgweir) [inline end-user/en/source/pages/apis/using-apis.rst]: ok

File: end-user/en/source/pages/apis/using-apis.rst (hunk 5 above)
Comment (jgweir) [inline end-user/en/source/pages/apis/using-apis.rst]: ok

File: end-user/en/source/pages/apis/using-apis.rst (hunk 6 above)
Comment (jgweir) [inline end-user/en/source/pages/apis/using-apis.rst]: ok

File: end-user/en/source/pages/apis/using-apis.rst (hunk 7 above)
Comment (jgweir) [inline end-user/en/source/pages/apis/using-apis.rst]: ok

File: end-user/en/source/pages/apis/using-apis.rst (hunk 8 above)
Comment (jgweir) [inline end-user/en/source/pages/apis/using-apis.rst]: yes good catch

**Pull request merged**